lib/data.py

```diff
@@ -24,47 +41,60 @@ def __init__(self, batch_id: str = "default"):
             config.MONGODB_PARAMS.collection
         )
         self.data = None
+        self.date_min = date_min
+        self.date_max = date_max
+        self.fields = fields
+        self.sample_size = sample_size
         self.batch_id = config.BATCH_ID if batch_id == "default" else batch_id
+        self.min_effectif = (
+            config.MIN_EFFECTIF if min_effectif == "default" else min_effectif
+        )
+        self.sirets = sirets
         self.mongo_pipeline = MongoDBQuery()
 
-    def fetch_data(
-        self,
-        date_min: str = "1970-01-01",
-        date_max: str = "3000-01-01",
-        fields: List = None,
-        sample_size: int = 0,  # a sample size of 0 means all data is retrieved
-        **kwargs,
-    ):
+    @classmethod
+    def from_config_file(cls, path: str, mode: str = "train"):
+        """
+        Instantiate a SFDataset object via a yaml config file
+        Args:
+            path: path to config file (typically in ./models/{version}/model.yml)
```

Comment:
Suggestion: reposer sur les commits du dépôt `git` pour le versionning => faire évoluer un unique fichier, plutot qu'accumuler des fichiers pour chaque version, dans des sous-répertoires.

```suggestion
            path: path to config file (typically in ./models/model.yml)
```